gc/base/standard/ParallelGlobalGC.cpp

```diff
@@ -185,6 +185,40 @@ fixObject(OMR_VMThread *omrVMThread, MM_HeapRegionDescriptor *region, omrobjectp
 	}
 }
 
+struct HeapSizes {
+	uintptr_t freeBytes;
+	uintptr_t objectBytes;
+};
+
+static void
+clearFreeEntry(OMR_VMThread *omrVMThread, MM_HeapRegionDescriptor *region, omrobjectptr_t object, void *userData)
+{
+	MM_GCExtensionsBase *extensions = MM_GCExtensionsBase::getExtensions(omrVMThread->_vm);
+	MM_ParallelGlobalGC *collector = (MM_ParallelGlobalGC *)extensions->getGlobalCollector();
+	HeapSizes *counter = (HeapSizes *)userData;
+
+	if (extensions->objectModel.isDeadObject(object)) {
+		if (extensions->objectModel.isSingleSlotDeadObject(object)) {
+			counter->freeBytes += extensions->objectModel.getSizeInBytesSingleSlotDeadObject(object);
+		} else {
+			/* Clear both linked and unlinked free entries. */
+			memset(
+				  (void *)((uintptr_t)object + sizeof(MM_HeapLinkedFreeHeader)),
+				  0,
+				  extensions->objectModel.getSizeInBytesMultiSlotDeadObject(object) - sizeof(MM_HeapLinkedFreeHeader));
+
+			counter->freeBytes += extensions->objectModel.getSizeInBytesMultiSlotDeadObject(object);
+		}
+	} else {
+		/* Marked objects are generally ignored except that assertions are
+		 * made that everything that is not a marked object is clearered, and
+		 * all items of the heap are visited (that the sum of sizes of all
+		 * marked objects and holes is equal to the heap size)
```

Comment:
slight rewrite and fixed a typo. does it retain its intended meaning?
```suggestion
		/* Marked objects are generally ignored; an assertion is used to
		 * ensure that unmarked objects are cleared; and all items of the
		 * heap are visited such that the sum of sizes of all marked
		 * objects and holes is equal to the heap size.
```

Reply:
I've separated this comment into two since the other assertion is inside the MM_ParallelGlobalGC::clearHeap()

Reply:
Should be fixed now